fennel/connectors/connectors.py

```diff
@@ -112,17 +112,25 @@ def decorator(dataset_cls: T):
 
 
 def sink(
-    conn: DataConnector, every: Optional[Duration] = None
+    conn: DataConnector,
+    cdc: str,
+    tier: Optional[Union[str, List[str]]] = None,
 ) -> Callable[[T], Any]:
+    if not isinstance(conn, DataConnector):
```

Comment:
Let's add restrictions on what do we allow as of today to be defined as sinks? 

We should ideally do that so that the user gets errors as part of "commit" without communication with the backend at all. This will allow the code paths to be tested even with mock clients which is the default local development workflow of our users

Reply:
I see that we have restriction on Kafka Connector as the sink at the proto conversion layer, but may be we should make it more explicit at the sink creator/decorator itself? Also we need to ensure the cdc conversion because proto conversion code allows literally any cdc method as of right now